### PR TITLE
[MBL-15693][Student] Course announcement notifications

### PR DESCRIPTION
**Pull request by kristofnemere**
refs: MBL-15693
affects: Student
release note: Added notification count badge.

test plan: Smoke test all kind of notification.

## Checklist

- [x] Follow-up e2e test ticket created or not needed
- [x] A11y checked
- [x] Approve from product or not needed

### CODE REVIEW

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by kdeakinstructure — CHANGES_REQUESTED** [review]
The notifications only marked as read after manual update. They shouldn't be bold after the user view them and navigates back to the notification list. (Also, the badge is only decreasing after manual refresh). We could investigate this together, because sometimes it is working, sometimes not, so it's not really consistent yet.

File: apps/student/src/main/java/com/instructure/student/fragment/NotificationListFragment.kt [hunk 1]
```diff
@@ -216,6 +242,8 @@ class NotificationListFragment : ParentFragment(), Bookmarkable {
                 SUBMISSION -> {
                     if (canvasContext !is Course) return
 
+                    markSubmissionAsRead(streamItem)
```
Comment (tamaskozmer) [inline apps/student/src/main/java/com/instructure/student/fragment/NotificationListFragment.kt:245]: I would move this into the submission screen or at least perform a network check here. It is possible that the user clicks the item, but it fails to load and we still mark it as read.

File: libs/pandares/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1283,6 +1283,11 @@
         <item quantity="other">%s unread messages</item>
     </plurals>
 
+    <plurals name="a11y_notificationsUnreadCount">
+        <item quantity="one">%s unread notification</item>
+        <item quantity="other">%s unread notification</item>
```
Comment (tamaskozmer) [inline libs/pandares/src/main/res/values/strings.xml:1288]: This should be `%s unread notifications`.

**Review by tamaskozmer — CHANGES_REQUESTED** [review]

**Review by kdeakinstructure — APPROVED** [review]
QA 👍

File: libs/pandares/src/main/res/values/strings.xml (hunk 2 above)
Comment (kristofnemere) [reply libs/pandares/src/main/res/values/strings.xml]: Fixed, thank you.

File: apps/student/src/main/java/com/instructure/student/fragment/NotificationListFragment.kt (hunk 1 above)
Comment (kristofnemere) [reply apps/student/src/main/java/com/instructure/student/fragment/NotificationListFragment.kt]: Fixed, thank you.

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Review by tamaskozmer — APPROVED** [review]

**Comment by inst-danger** [comment]
<p>Teacher Install Page</p>

**Comment by inst-danger** [comment]
<p>Student Install Page</p>

**Pull request merged**